rosidl_generator_py/resource/_msg_support.c.template

```diff
@@ -123,11 +130,22 @@ PyObject * @(spec.base_type.pkg_name)_@(module_name)__convert_to_py(void * raw_r
 @[for field in spec.fields]@
   PyObject * py@(field.name) = NULL;
 @[if not field.type.is_primitive_type()]@
-// TODO(esteve): add support for nested types
-  (void)py@(field.name);
-@[end if]@
-
+@{
+nested_type = '%s__%s__%s' % (field.type.pkg_name,'msg',field.type.type)
+}@
 @[if field.type.is_array]@
+  // TODO(mikael) implement nested array
+  assert(PySequence_Check(py@(field.name)));
+@[else]@
+  PyObject * py@(field.name)_msg_module = PyImport_ImportModule("@(field.type.pkg_name).msg._@convert_camel_case_to_lower_case_underscore(field.type.type)");
+  PyObject* py@(field.name)_convert_to_py = PyObject_GetAttrString(py@(field.name)_msg_module, "_convert_to_py");
+  typedef PyObject *(* convert_to_py_signature)(void *);
+  convert_to_py_signature convert_to_py_@(field.name) = (convert_to_py_signature)PyCapsule_GetPointer(py@(field.name)_convert_to_py,NULL);
+  @(nested_type) pytmp@(field.name) = ros_message->@(field.name);
+  py@(field.name) = convert_to_py_@(field.name)(&pytmp@(field.name));
+@[end if]@
+@[elif field.type.is_array]@
```

Comment:
Is this template currently working? From reading I can't see to which this `elif` belongs to. But I might just miss it.